### DISCUSSION TITLE
deviceSerialNumber

### DISCUSSION THREAD

**Discussion by pt-at-tg** [christianrowlands/android-network-survey#53] [General]
Hi Christian, on the mqtt backend in the msg payload there is a key-val for deviceSerialNumber.  Since getting the device's serial number is basically impossible after android 10, some Qs:

Is the val generated by the app? 
How persistent is it?

Thanks,
Peter

**Comment by christianrowlands** [comment]
You are correct that getting the real device SN is practically impossible by an app. Therefore, I use the ANDROID_ID provided by the Android API to set the deviceSerialNumber. Back when I first wrote the app I was able to get the IMEI, but that has changed as you mentioned.

The documentation on the ANDROID_ID can be found here: https://developer.android.com/reference/android/provider/Settings.Secure#ANDROID_ID
